static/js/chart/drawChoropleth.ts

```diff
@@ -0,0 +1,260 @@
+        return colorScale(value);
+      } else {
+        return MISSING_DATA_COLOR;
+      }
+    })
+    .attr("id", (_, index) => {
+      return "geoPath" + index;
+    })
+    .on("mouseover", onMouseOver(container_id))
+    .on("mouseout", onMouseOut(container_id))
+    .on("mousemove", onMouseMove(container_id, dataValues, unit));
+
+  generateLegend(container_id, colorScale, unit);
+  addTooltip(container_id);
+}
+
+const onMouseOver = (container_id: string) => (_, index): void => {
+  // show highlighted border
+  d3.select("#geoPath" + index)
+    .classed("border", false)
+    .classed("border-highlighted", true);
+  // show tooltip
+  d3.select("#" + container_id + " #" + TOOLTIP_ID).style("display", "block");
+};
+
+const onMouseOut = (container_id: string) => (_, index): void => {
+  d3.select("#geoPath" + index)
+    .classed("border", true)
+    .classed("border-highlighted", false);
+  d3.select("#" + container_id + " #" + TOOLTIP_ID).style("display", "none");
+};
+
+const onMouseMove = (
+  container_id: string,
+  dataValues: { [placeDcid: string]: number },
+  unit: string
+) => (e) => {
+  const placeName = e["properties"].name;
+  const value = unit
+    ? dataValues[e["properties"].geoDcid] + unit
+    : dataValues[e["properties"].geoDcid];
+  const text = placeName + ": " + value;
+  d3.select("#" + container_id + " #tooltip")
+    .text(text)
+    .style("left", d3.event.offsetX + 3 + "px")
+    .style("top", d3.event.offsetY - 3 + "px");
+};
+
+function determineColorPalette(dataValues: {
+  [placeDcid: string]: number;
+}): number[] {
+  // Create a sorted list of values.
+  const values = [];
+  for (const key in dataValues) {
+    values.push(dataValues[key]);
+  }
+  values.sort((a, b) => a - b);
+  const len = values.length;
+
+  // Find CHOROPLETH_MIN_DATAPOINTS number of values with equal separation from one another.
+  const steps = CHOROPLETH_MIN_DATAPOINTS;
+  if (len >= steps) {
+    const start = 0;
+    return d3.range(start, steps).map((d) => {
+      return values[Math.floor(((len - 1) * d) / (steps - 1))];
+    });
+  } else {
+    return [0, 0, 0, 0, 0, 0, 0, 0, 0];
+  }
+}
+
+function addTooltip(container_id: string) {
+  d3.select("#" + container_id)
+    .append("div")
+    .attr("id", TOOLTIP_ID)
+    .attr("style", "position: absolute; display: none; z-index: 10");
+}
+
+/**
+ * Draw a color scale legend.
+ * @param color The d3 linearScale that encodes the color gradient to be
+ *        plotted.
+ */
+function generateLegend(
+  id: string,
+  color: d3.ScaleLinear<number, number>,
+  unit: string
+) {
+  const width = 250;
+  const height = 50;
+  const tickSize = 6;
+  const title = unit ? "Color Scale (" + unit + ")" : "Color Scale";
+  const marginTop = 18;
+  const marginBottom = 16 + tickSize;
+  const marginSides = 15;
+  const textPadding = 6;
+  const numTicks = 5;
+
+  d3.select("#" + id)
+    .append("div")
+    .attr("class", "choropleth-legend");
+
+  const svg = d3
```

Comment:
this is technically a div (also, no need to select for this since it was just create a few lines above)